### ISSUE TITLE
[mlir][TOSA] TOSA MLIR dialect does not properly verify broadcast compatibility

### ISSUE THREAD

**Issue by eric-k256** [llvm/llvm-project#61822]
TOSA ops have the ResultsBroadcastableShape, which adds a verifier to make sure that the shapes of the operands/results are appropriate with broadcasting. The problem is that TOSA broadcasting has stricter requirements (rank of both operands must be equal), which is not caught with the existing trait. The TOSA dialect likely needs its own verifier to use TOSA rules verifying broadcast compatibility.

**Comment by lhutton1** [comment]
Closing as this appears to be resolved:

```mlir
func.func @test_add(%arg0: tensor<13x21xf32>, %arg1: tensor<13x21x3xf32>) -> tensor<13x21x3xf32> {
  %0 = tosa.sub %arg0, %arg1 : (tensor<13x21xf32>, tensor<13x21x3xf32>) -> tensor<13x21x3xf32>
  return %0 : tensor<13x21x3xf32>
}

$ mlir-opt test.mlir
test.mlir:366:8: error: 'tosa.sub' op operands don't have broadcast-compatible shapes
  %0 = tosa.sub %arg0, %arg1 : (tensor<13x21xf32>, tensor<13x21x3xf32>) -> tensor<13x21x3xf32>
       ^
test.mlir:366:8: note: see current operation: %0 = "tosa.sub"(%arg0, %arg1) : (tensor<13x21xf32>, tensor<13x21x3xf32>) -> tensor<13x21x3xf32>
```
